conda_rpms/generate.py

```diff
@@ -49,17 +50,28 @@ def render_dist_spec(dist, config):
                                 rpm_prefix=rpm_prefix,
                                 install_prefix=install_prefix)
 
-def render_env(branch_name, label, repo, config, tag, commit_num):
+
+def render_env(branch_name, label, config, tag, commit_num):
+    install_prefix = config['install']['prefix']
+    rpm_prefix = config['rpm']['prefix']
     env_info = {'url': 'http://link/to/gh',
                 'name': branch_name,
-		'label' : label,
-                'summary': 'A SciTools environment.',
+                'label': label,
+                'summary': 'A {} environment.'.format(rpm_prefix),
                 'version': commit_num,}
-    install_prefix = config['install']['prefix']
-    rpm_prefix = config['rpm']['prefix']
+    # When multiple tags are produced in a day, they have an associated count
+    # addded to the end e.g. env-default-2016_12_05-2, which needs to be parsed
+    # correctly.
+    tag_pattern = re.compile('env\-[a-zA-Z]+\-([0-9\_\-]+)')
```

Comment:
@lbdreyer It's better to compile the regular expression pattern once and then reuse it multiple times, otherwise there's no point in using this optimisation.

I'd suggest breaking this into two parts of compiling the pattern at the module level, and then use the pattern for matching within this function.

Note that, we should also make the tag name alphanumeric rather than restrict it to just being composed of alphabetic characters, so I'd suggest the following regular expression pattern

```python
import re

TAG_PATTERN = '^env-\w+-(\d{4}_\d{2}_\d{2}(-\d+)?)$'
tag_pattern = re.compile(PATTERN)
```